cli/command/trust/key_load.go

```diff
@@ -0,0 +1,94 @@
+package trust
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary"
+	"github.com/docker/notary/storage"
+	"github.com/docker/notary/trustmanager"
+	tufutils "github.com/docker/notary/tuf/utils"
+	"github.com/spf13/cobra"
+)
+
+const (
+	ownerReadOnlyPerms     = 0400
+	ownerReadAndWritePerms = 0600
+)
+
+type keyLoadOptions struct {
+	keyName string
+}
+
+func newKeyLoadCommand(dockerCli command.Streams) *cobra.Command {
+	var options keyLoadOptions
+	cmd := &cobra.Command{
+		Use:   "load [OPTIONS] KEY",
+		Short: "Load a private key file for signing",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return loadPrivKey(dockerCli, args[0], options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&options.keyName, "name", "n", "signer", "Name for the loaded key")
+	return cmd
+}
+
+func loadPrivKey(streams command.Streams, keyPath string, options keyLoadOptions) error {
+	trustDir := trust.GetTrustDirectory()
+	keyFileStore, err := storage.NewPrivateKeyFileStorage(trustDir, notary.KeyExtension)
+	if err != nil {
+		return err
+	}
+	privKeyImporters := []trustmanager.Importer{keyFileStore}
+
+	fmt.Fprintf(streams.Out(), "\nLoading key from \"%s\"...\n", keyPath)
+
+	// Always use a fresh passphrase retriever for each import
+	passRet := trust.GetPassphraseRetriever(streams.In(), streams.Out())
+	keyBytes, err := getPrivKeyBytesFromPath(keyPath)
+	if err != nil {
+		return fmt.Errorf("error reading key from %s: %s", keyPath, err)
+	}
+	if err := loadPrivKeyBytesToStore(keyBytes, privKeyImporters, keyPath, options.keyName, passRet); err != nil {
+		return fmt.Errorf("error importing key from %s: %s", keyPath, err)
+	}
+	fmt.Fprintf(streams.Out(), "Successfully imported key from %s\n", keyPath)
+	return nil
+}
+
+func getPrivKeyBytesFromPath(keyPath string) ([]byte, error) {
+	fileInfo, err := os.Stat(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	if fileInfo.Mode() != ownerReadOnlyPerms && fileInfo.Mode() != ownerReadAndWritePerms {
+		return nil, fmt.Errorf("private key permission from %s should be set to 400 or 600", keyPath)
```

Comment:
What about 700, or 200? Maybe this could say "private key must not be readable by others" or something like that? 

I think the ssh client says something like this: "It is required that your private key files are NOT accessible by others."

Reply:
I don't think 200 or 700 make sense here, for 200 you'd have writable but not readable (I've never seen this done?), and with 700 the key would be executable (also don't see a use for this?).

Reply:
I agree those would be silly permissions. My point is that the error message is misleading. The requirement is that it can't be readable by others, not that it must be one of those two permissions. The error message should match the requirement. The ssh client is a good example of this.

If I'm not mistaken this error message will also be used on windows. "set to 400 or 600" is meaningless to windows users.

Reply:
Ah, I just noticed the code *does* actually check for specifically those permissions. I think this needs to be changed to just check that the specific bits are not set.

Reply:
sounds good, updated!